src/Pos/Delegation/Logic.hs

```diff
@@ -173,20 +176,47 @@ getProxyMempool
     :: (MonadDB m, MonadDelegation m)
     => m ([ProxySKHeavy], [ProxySKHeavy])
 getProxyMempool = do
-    sks <- runDelegationStateAction $ ether $
+    sks <- runDelegationStateAction $
         uses dwProxySKPool HM.elems
     let issuers = map pskIssuerPk sks
     toRollback <- catMaybes <$> mapM GS.getPSKByIssuer issuers
     pure (sks, toRollback)
 
-clearProxyMemPool
+clearDlgMemPool
     :: (MonadDB m, MonadDelegation m)
     => m ()
-clearProxyMemPool = do
-    var <- askDelegationState
-    atomically $ modifyTVar' var resetData
-  where
-    resetData delegationWrap = delegationWrap {_dwProxySKPool = mempty}
+clearDlgMemPool = runDelegationStateAction clearDlgMemPoolAction
+
+clearDlgMemPoolAction :: DelegationStateAction ()
+clearDlgMemPoolAction = do
+    dwProxySKPool .= mempty
+    dwPoolSize .= 1
+
+-- Put value into Proxy SK Pool. Value must not exist in pool.
```

Comment:
Why shouldn't it exist? It's alright to override the proxy certificate -- user can issue cert and then change his mind and issue another one, thus second will be put into blockchain instead of the first one => profit.